lua_cjson.c

```diff
@@ -1399,7 +1412,10 @@ static void json_process_value(lua_State *l, json_parse_t *json,
     case T_NULL:
         /* In Lua, setting "t[k] = nil" will delete k from the table.
          * Hence a NULL pointer lightuserdata object is used instead */
-        lua_pushlightuserdata(l, NULL);
+        if (json->cfg && json->cfg->decode_null_as_nil)
```

Comment:
json_process_value will also be called when parsing the array. Will this cause `[null, 1]`  become `{[2]=1}` or an array with nil hole?

Reply:
@spacewander updated